crates/sui-graphql-client/src/lib.rs

```diff
@@ -539,6 +606,139 @@ impl Client {
             .map(|c| c.sequence_number))
     }
 
+    // ===========================================================================
+    // Dynamic Field(s) API
+    // ===========================================================================
+
+    /// Access a dynamic field on an object using its name. Names are arbitrary Move values whose
+    /// type have copy, drop, and store, and are specified using their type, and their BCS
+    /// contents, Base64 encoded.
+    ///
+    /// The `name` argument can be either a [`BcsName`] for passing raw bcs bytes or a type that
+    /// implements Serialize.
+    ///
+    /// This returns [`DynamicFieldOutput`] which contains the name, the value as json, and object.
+    ///
+    /// # Example
+    /// ```rust,ignore
+    ///
+    /// let client = sui_graphql_client::Client::new_devnet();
+    /// let address = Address::from_str("0x5").unwrap();
+    /// let df = client.dynamic_field_with_name(address, "u64", 2u64).await.unwrap();
+    ///
+    /// # alternatively, pass in the bcs bytes
+    /// let bcs = base64ct::Base64::decode_vec("AgAAAAAAAAA=").unwrap();
+    /// let df = client.dynamic_field(address, "u64", BcsName(bcs)).await.unwrap();
+    /// ```
+    pub async fn dynamic_field(
+        &self,
+        address: Address,
+        type_: &str,
+        name: impl Into<NameValue>,
+    ) -> Result<Option<DynamicFieldOutput>, Error> {
+        let bcs = name.into().0;
+        let operation = DynamicFieldQuery::build(DynamicFieldArgs {
+            address,
+            name: crate::query_types::DynamicFieldName {
+                type_: type_.to_string(),
```

Comment:
nit: if the body of this function ultimately needs to take ownership of the `type_`, then IMO it's preferable to indicate that in the type signature by accepting `type_: String`, otherwise in some scenarios you are forcing an unnecessary copy.

Reply:
Do we want to use the TypeTag type directly instead of Strings?